gems/parallel/1.20/parallel.rbs

```diff
@@ -0,0 +1,83 @@
+                         ?in_threads: Integer,
+                         ?progress: _ToStr,
+                         ?start: ^(T item, Integer index) -> untyped,
+                         ?finish: ^(T item, Integer index, U result) -> untyped) { (T) -> U } -> Enumerable[U]
+               | [T, U] ((^() -> T | untyped) array,
+                         ?in_processes: Integer,
+                         ?in_threads: Integer,
+                         ?progress: _ToStr,
+                         ?start: ^(T item, Integer index) -> untyped,
+                         ?finish: ^(T item, Integer index, U result) -> untyped) { (T) -> U } -> (^() -> T | untyped)
+
+  def self.each_with_index: [T, U] (Enumerable[T] array,
+                                    ?in_processes: Integer,
+                                    ?in_threads: Integer,
+                                    ?progress: _ToStr,
+                                    ?start: ^(T item, Integer index) -> untyped,
+                                    ?finish: ^(T item, Integer index, U result) -> untyped) { (T, Integer) -> U } -> Enumerable[T]
+                          | [T, U] ((^() -> T | untyped) array,
+                                    ?in_processes: Integer,
+                                    ?in_threads: Integer,
+                                    ?progress: _ToStr,
+                                    ?start: ^(T item, Integer index) -> untyped,
+                                    ?finish: ^(T item, Integer index, U result) -> untyped) { (T, Integer) -> U } -> (^() -> T | untyped)
+
+  def self.flat_map: [T, U] (Enumerable[T] array,
+                             ?in_processes: Integer,
+                             ?in_threads: Integer,
+                             ?progress: _ToStr,
+                             ?start: ^(T item, Integer index) -> untyped,
+                             ?finish: ^(T item, Integer index, U result) -> untyped) { (T) -> U } -> Array[U]
+
+  def self.map: [T, U] (Enumerable[T] | (^() -> T | untyped) array,
+                        ?in_processes: Integer,
+                        ?in_threads: Integer,
+                        ?progress: _ToStr,
+                        ?start: ^(T item, Integer index) -> untyped,
+                        ?finish: ^(T item, Integer index, U result) -> untyped) { (T) -> U } -> Array[U]
+
+  def self.map_with_index: [T, U] (Enumerable[T] | (^() -> T | untyped) array,
+                                   ?in_processes: Integer,
+                                   ?in_threads: Integer,
+                                   ?progress: _ToStr,
+                                   ?start: ^(T item, Integer index) -> untyped,
+                                   ?finish: ^(T item, Integer index, U result) -> untyped) { (T, Integer) -> U } -> Array[U]
+
+  def self.physical_processor_count: () -> Integer
+
+  def self.processor_count: () -> Integer
+
+  def self.worker_number: () -> Integer
```

Comment:
[note] When evaluating `Parallel.worker_number` without a block, it returns `nil`:

```
irb(main):003:0> Parallel.worker_number
=> nil
```

But this usage is incorrect and we should not want nil-check for `Parallel.worker_number`, so I decide to return `Integer` instead of `Integer?`.

If the return type would be `Integer?`, the following code would encounter a type error:

```
number.rb:1:63: [error] Type `(::Integer | nil)` does not have method `/`
│ Diagnostic ID: Ruby::NoMethod
│
└ Parallel.each(1..5) { |i| puts "#{i}: #{Parallel.worker_number / 2.0}" }
                                                                 ~
```